source/kraken/maintenance_worker.cpp

```diff
@@ -82,69 +85,110 @@ void MaintenanceWorker::operator()(){
     }
 }
 
-void MaintenanceWorker::handle_task(AmqpClient::Envelope::ptr_t envelope){
-    LOG4CPLUS_TRACE(logger, "Task received");
-    pbnavitia::Task task;
-    bool result = task.ParseFromString(envelope->Message()->Body());
-    if(!result){
-        LOG4CPLUS_WARN(logger, "Protobuf is not valid!");
-        return;
-    }
-    switch(task.action()){
-        case pbnavitia::RELOAD:
-            load(); break;
-        default:
-            LOG4CPLUS_TRACE(logger, "Task ignored");
+void MaintenanceWorker::handle_task_in_batch(const std::vector<AmqpClient::Envelope::ptr_t>& envelopes){
+    for (auto& envelope: envelopes) {
```

Comment:
I agree with you totally. In this case we can pass a vector that has only one task. Or, I can do a wrapper that takes only one message and delegate to this function?